elastic/kibana
# [Security Solution] Ensure alerts are scheduled when rule times out

## Summary
Fixes: https://github.com/elastic/kibana/issues/121559

Related:
https://github.com/elastic/kibana/pull/120506

NOTE: Tests are very difficult to write for this case. There are tests in the alerting framework for the constructs used. Below is the method I used for manual testing, in detail.

To test:
1. Start an http server on localhost:
`sudo python3 -m http.server 5605` or use `netcat`.

2. Create a rule that creates alerts and then times out. Easiest way I have found to accomplish this is below:
<img width="889" alt="image" src="https://user-images.githubusercontent.com/611653/159722285-a02a055e-dee2-455c-92ec-e13e53df4796.png">
<img width="935" alt="image" src="https://user-images.githubusercontent.com/611653/159722600-25f24d5c-7796-466e-9c3b-d5dacc11b9a1.png">
<img width="624" alt="image" src="https://user-images.githubusercontent.com/611653/159723590-cf9167fa-a887-4735-afc6-670787d9d54c.png">
If you create more than 2 alerts, the first 2 alerts will be created and then the task will stall, and eventually time out.<br/><br/>

3. Check rule status after time out. You should see alerts created and rule notifications should fire for those 2 alerts (in example below, POST is not supported by the simple Python webserver, but you can see that the notification action did fire)
<img width="1308" alt="image" src="https://user-images.githubusercontent.com/611653/159723099-d0ad5fd0-bfe2-4326-b017-95cba263d8aa.png">
<img width="737" alt="image" src="https://user-images.githubusercontent.com/611653/159723670-85ae1cd1-9817-48cf-8d7d-8afc9aeac2c1.png">




### For maintainers
